libafl_bolts/src/anymap.rs

```diff
@@ -39,34 +39,21 @@ macro_rules! impl_asany {
     };
 }
 
-/// Get a `type_id` from its previously unpacked `u64`.
+/// Get a `type_id` from its previously unpacked `u128`.
 /// Opposite of [`unpack_type_id(id)`].
 ///
 /// # Note
 /// Probably not safe for future compilers, fine for now.
 /// The size changed in later rust versions, see <https://github.com/rust-lang/compiler-team/issues/608>
 #[inline]
 #[must_use]
-#[allow(clippy::cast_ptr_alignment)]
 pub const fn pack_type_id(id: u128) -> TypeId {
-    match size_of::<TypeId>() {
-        8 => {
-            let id_64 = id as u64;
-            // false positive: this branch only executes on 64 bit `TypeId`s
-            #[allow(clippy::cast_ptr_alignment)]
-            unsafe {
-                *(addr_of!(id_64) as *const TypeId)
-            }
-        }
-        16 => unsafe { *(addr_of!(id) as *const TypeId) },
-        _ => {
-            // TypeId size of this size is not yet supported"
-            panic!("Unsupported size for TypeId");
-        }
-    }
+    // TypeId size of other sizes is not yet supported"
+    assert!(size_of::<TypeId>() == 16, "Unsupported size for TypeId");
```

Comment:
Could be `assert_eq!` (and could even be checked at compile time with something like [static_assertions](https://docs.rs/static_assertions/latest/static_assertions/) althoug it doesn't matter)